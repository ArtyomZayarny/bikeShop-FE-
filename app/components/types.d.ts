export type ListItem = {
  id: string;
};

export interface LinkItem extends ListItem {
  href: string;
  name: string;
}

export type filterItem = {
  id: string;
  label: string;
};

export interface filterBlock {
  id: string;
  label: string;
  type: string;
  items?: (filerItem | string)[];
  maxPrice?: number;
}

export interface IProduct {
  id: string;
  title: string;
  discount: number;
  imageUrl: string;
  inStock: boolean;
  price: PriceType;
  attributes: filterBlock[];
  href: string;
}

export type PriceType = {
  newPrice: number;
  oldPrice: number;
};
