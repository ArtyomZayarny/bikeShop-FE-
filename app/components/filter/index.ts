export * from "./quick-filter-btn/quick-filter-btn";
export * from "./filter";
