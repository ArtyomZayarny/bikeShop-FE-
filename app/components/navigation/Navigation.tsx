import Image from "next/image";
import styles from "./styles.module.css";
import Link from "next/link";
import { Container } from "../container/container";

export const Navigation = () => {
  return (
    <nav id="navigation" className={`flex justify-center ${styles.navigation}`}>
      <Container styles="items-center md:grid md:grid-cols-2 md:grid-row-1 md:gap-y-4">
        {/* Logo */}
        <div className="logo">
          <Link href="/">
            <Image src="./svg/logo.svg" alt="logo" width={200} height={206} />
          </Link>
        </div>
        {/* Main navigation */}
        <ul className="flex gap-6 md:justify-end">
          <li>
            <div className="flex">
              <Image
                src="./svg/burger.svg"
                alt="catalog"
                width={20}
                height={20}
              />
              <Link className="text-color font-[600]" href="#catalog">
                Каталог
              </Link>
            </div>
          </li>
          <li>
            <Link className="text-color font-[600]" href="#new">
              Новинки
            </Link>
          </li>
          <li>
            <Link className="accent font-[600]" href="#sale">
              Акції
            </Link>
          </li>
          <li>
            <Link className="text-color font-[600]" href="#services">
              Послуги
            </Link>
          </li>
        </ul>

        {/* Searh form */}
        <div className="w-full md:w-full md:col-span-2 lg:max-w-sm ">
          <form
            action=""
            className="flex items-center lg:max-w-md bg-white rounded-full px-6 py-3 w-full md:w-full"
          >
            <input type="text" placeholder="" className="outline-none w-full" />
            <Image src="./svg/search.svg" alt="search" width={24} height={24} />
            <button type="submit" hidden>
              Search
            </button>
          </form>
        </div>
      </Container>
    </nav>
  );
};
