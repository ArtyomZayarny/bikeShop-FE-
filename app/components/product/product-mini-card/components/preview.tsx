import Image from "next/image";

type Props = {
  imageUrl: string;
};

export const Preview = ({ imageUrl }: Props) => {
  return (
    <div className="flex justify-center">
      <Image src={imageUrl} alt="trek" width={200} height={107} />
    </div>
  );
};
