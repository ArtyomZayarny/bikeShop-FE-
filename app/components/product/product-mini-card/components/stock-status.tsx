type Props = {
  inStock: boolean;
};
export const StockStatus = ({ inStock }: Props) => {
  return (
    <div>
      <span
        className={`text-sm font-normal text-${inStock ? "green" : "disabled"}`}
      >
        {inStock ? "В наявності" : "Немає в наявності"}
      </span>
    </div>
  );
};
