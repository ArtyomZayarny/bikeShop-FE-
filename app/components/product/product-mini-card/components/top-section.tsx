import Image from "next/image";
import styles from "../../styles.module.css";

type Props = {
  discount: number;
};

export const TopSection = ({ discount }: Props) => {
  return (
    <div className="flex justify-between">
      {/* discount */}
      <div>
        <span className={`text-redAccent text-sm font-semibold`}>
          -{discount}%
        </span>
      </div>
      {/* actions buttons */}
      <div className={`${styles.actionsBtn}`}>
        <div className="flex gap-2">
          <Image
            src="./svg/heart.svg"
            alt="like-button"
            width={24}
            height={24}
          />
          <Image
            src="./svg/balance.svg"
            alt="add-to-compare-list"
            width={24}
            height={24}
          />
        </div>
      </div>
    </div>
  );
};
