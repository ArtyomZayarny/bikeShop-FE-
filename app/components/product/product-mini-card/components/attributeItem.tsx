import { DropDown } from "../../../common/dropdown";
import { Title } from "../../../common/title";
import { ColorList } from "../../../filter/color-list";
import { filterBlock } from "../../../types";

type Props = {
  attr: filterBlock;
};
export const AttributeItem = ({ attr }: Props) => {
  const renderAttributeItem = () => {
    switch (attr.type) {
      case "color":
        return (
          <>
            <ColorList colors={attr.items!} />
          </>
        );
      case "size":
        return (
          <>
            <DropDown options={attr.items!} />
          </>
        );
    }
  };
  return (
    <li className="flex mb-4 flex-col">
      <Title type={6} styles="text-sm font-normal mb-2">
        {attr.label}
      </Title>
      {renderAttributeItem()}
    </li>
  );
};
