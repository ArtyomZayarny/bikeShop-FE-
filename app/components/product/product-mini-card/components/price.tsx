import { PriceType } from "@/app/components/types";

type Props = {
  price: PriceType;
};

export const Price = ({ price }: Props) => {
  return (
    <div>
      {/* price */}
      <span className="text-xs text-regular font-semibold">
        {price.newPrice} грн
      </span>
      {/* old price */}
      <span className="text-xs ml-2 text-disabled line-through">
        {price.oldPrice} грн
      </span>
    </div>
  );
};
