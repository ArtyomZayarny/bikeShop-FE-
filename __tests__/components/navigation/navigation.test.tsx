import { render } from "@testing-library/react";
import { Navigation } from "../../../app/components/navigation";

describe("Navigation", () => {
  it("should render Navigation", () => {
    const { container } = render(<Navigation />);
    expect(container).toMatchSnapshot();
  });
});
