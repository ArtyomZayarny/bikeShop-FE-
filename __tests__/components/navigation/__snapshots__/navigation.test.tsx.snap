// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render Navigation 1`] = `
<div>
  <nav
    class="flex justify-center navigation"
    id="navigation"
  >
    <div
      class="justify-between py-6 w-full 2xl:max-w-screen-2xl xl:max-w-screen-xl lg:px-10 lg:flex md:px-5 items-center md:grid md:grid-cols-2 md:grid-row-1 md:gap-y-4"
    >
      <div
        class="logo"
      >
        <a
          href="/"
        >
          <img
            alt="logo"
            data-nimg="1"
            decoding="async"
            height="206"
            loading="lazy"
            src="./svg/logo.svg"
            style="color: transparent;"
            width="200"
          />
        </a>
      </div>
      <ul
        class="flex gap-6 md:justify-end"
      >
        <li>
          <div
            class="flex"
          >
            <img
              alt="catalog"
              data-nimg="1"
              decoding="async"
              height="20"
              loading="lazy"
              src="./svg/burger.svg"
              style="color: transparent;"
              width="20"
            />
            <a
              class="text-color font-[600]"
              href="#catalog"
            >
              Каталог
            </a>
          </div>
        </li>
        <li>
          <a
            class="text-color font-[600]"
            href="#new"
          >
            Новинки
          </a>
        </li>
        <li>
          <a
            class="accent font-[600]"
            href="#sale"
          >
            Акції
          </a>
        </li>
        <li>
          <a
            class="text-color font-[600]"
            href="#services"
          >
            Послуги
          </a>
        </li>
      </ul>
      <div
        class="w-full md:w-full md:col-span-2 lg:max-w-sm "
      >
        <form
          action=""
          class="flex items-center lg:max-w-md bg-white rounded-full px-6 py-3 w-full md:w-full"
        >
          <input
            class="outline-none w-full"
            placeholder=""
            type="text"
          />
          <img
            alt="search"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="./svg/search.svg"
            style="color: transparent;"
            width="24"
          />
          <button
            hidden=""
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
    </div>
  </nav>
</div>
`;
